### ISSUE TITLE
Hide "real" caret during visual mode

### ISSUE THREAD

**Issue by musjj** [VSCodeVim/Vim#9007]
**Is your feature request related to a problem? Please describe.**
It looks like during normal mode, the plugin simulates a fake(?) caret, but the real caret stays visible:

![image](https://github.com/VSCodeVim/Vim/assets/72612857/48cdea3a-14d9-421d-9bf0-a59edc17cbef)

With `editor.cursorSmoothCaretAnimation` set to `on`, this becomes really apparent and distracting:

[Kooha-2024-04-26-23-09-36.webm](https://github.com/VSCodeVim/Vim/assets/72612857/78186a65-18b4-4da1-8331-38b44e18e8c7)

**Describe the solution you'd like**
I believe that this could be solved by simply temporarily making the "real" caret invisible, by setting it to a transparent color. The color can then be reset afterwards.

**Describe alternatives you've considered**
I tried to figure out a way to hack it from the user's end, but I couldn't figure it out.

**Comment by J-Fields** [comment]
> I believe that this could be solved by simply temporarily making the "real" caret invisible, by setting it to a transparent color. The color can then be reset afterwards.

I don't think this is feasible. It would require modifying settings every time we enter/exit Visual mode - this is quite disruptive, as users of `statusBarColors` know.
